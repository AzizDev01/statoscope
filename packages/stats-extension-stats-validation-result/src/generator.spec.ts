import { ExtensionDescriptor } from '@statoscope/stats/spec/extension';
import Generator from './generator';

const adapter: ExtensionDescriptor = { name: 'test-adapter', version: '7.7.7' };

test('should work', () => {
  const generator = new Generator(adapter);

  generator.handleRule('foo-rule', {
    description: 'foo-description',
    package: {
      name: 'foo-package',
      version: '1.0.0',
      homepage: 'https://statoscope.tech',
    },
  });

  generator.handleEntry('foo-rule', {
    type: 'error',
    message: 'foo-message',
    compilation: 'foo-compilation',
    related: [
      { type: 'module', id: 'foo-foo-module' },
      { type: 'entry', id: 'foo-foo-entry' },
    ],
  });

  generator.handleEntry('foo-rule', {
    type: 'error',
    message: 'bar-message',
    compilation: 'foo-compilation',
    related: [
      { type: 'module', id: 'foo-foo-module' },
      { type: 'module', id: 'foo-bar-module' },
      { type: 'entry', id: 'foo-foo-entry' },
      { type: 'entry', id: 'foo-bar-entry' },
    ],
  });

  generator.handleEntry('foo-rule', {
    type: 'error',
    message: 'bar-message',
    compilation: 'bar-compilation',
    related: [
      { type: 'module', id: 'bar-foo-module' },
      { type: 'module', id: 'bar-bar-module' },
      { type: 'entry', id: 'bar-foo-entry' },
      { type: 'entry', id: 'bar-bar-entry' },
    ],
  });

  expect(generator.get()).toMatchSnapshot();
});
