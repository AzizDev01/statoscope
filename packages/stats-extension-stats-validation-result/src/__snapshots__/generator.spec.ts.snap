// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should work 1`] = `
Object {
  "descriptor": Object {
    "adapter": Object {
      "name": "test-adapter",
      "version": "7.7.7",
    },
    "author": Object {
      "name": "Sergey Melyukov",
      "url": "https://twitter.com/smelukov",
    },
    "homepage": "https://github.com/statoscope/statoscope/",
    "name": "@statoscope/stats-extension-stats-validation-result",
    "version": "5.6.1",
  },
  "payload": Object {
    "compilations": Array [
      Object {
        "id": "foo-compilation",
        "items": Array [
          Object {
            "details": Array [],
            "id": 0,
            "message": "foo-message",
            "related": Array [
              Object {
                "id": "foo-foo-module",
                "type": "module",
              },
              Object {
                "id": "foo-foo-entry",
                "type": "entry",
              },
            ],
            "rule": "foo-rule",
            "type": "error",
          },
          Object {
            "details": Array [],
            "id": 1,
            "message": "bar-message",
            "related": Array [
              Object {
                "id": "foo-foo-module",
                "type": "module",
              },
              Object {
                "id": "foo-bar-module",
                "type": "module",
              },
              Object {
                "id": "foo-foo-entry",
                "type": "entry",
              },
              Object {
                "id": "foo-bar-entry",
                "type": "entry",
              },
            ],
            "rule": "foo-rule",
            "type": "error",
          },
        ],
      },
      Object {
        "id": "bar-compilation",
        "items": Array [
          Object {
            "details": Array [],
            "id": 2,
            "message": "bar-message",
            "related": Array [
              Object {
                "id": "bar-foo-module",
                "type": "module",
              },
              Object {
                "id": "bar-bar-module",
                "type": "module",
              },
              Object {
                "id": "bar-foo-entry",
                "type": "entry",
              },
              Object {
                "id": "bar-bar-entry",
                "type": "entry",
              },
            ],
            "rule": "foo-rule",
            "type": "error",
          },
        ],
      },
    ],
  },
}
`;
